jdk17-unit-tests/src/test/java/com/uber/nullaway/jdk17/NullAwayModuleInfoTests.java

```diff
@@ -36,6 +36,7 @@ public void testModuleInfo() {
         .addSourceLines(
             "module-info.java",
             "module com.uber.mymodule {",
+            "  requires java.base;",
```

Comment:
The difference here isn't `static` vs not, but just that this one is a chain of two derefs so the "base expression" is just `java` which is an identifier rather than a member select expression. Correct?

Reply:
Yes.  For the previous case `require static org.checkerframework.checker.qual;`, `matchMemberSelect()` would get called with the sub-expression `org.checkerframework`, but since the symbol for that sub-expression was null, `matchDeference()` was not called for identifier `org`.  `java.base` has a non-null `Symbol` representing the module, so `matchDereference()` was being called with identifier `java`, leading to the new crash.

Reply:
Maybe here it's worth having a small comment:
```
// Important: two-level deep module tests matching of identifier `java` (see #544)
```
Not sure how I feel about a PR's code referring to itself, but a full explanation might be too verbose.